### PR TITLE
Improve navigation

### PR DESCRIPTION
**Pull request by paulsonnentag**
- Add useViewState hook that allows components to store state that's addressable through the URL
- Remove URL state from workspace
- Remove special forward/backwards/copy-url buttons
- Each user stores their current URL in the top level document that they are currently viewing
   - this allows to jump to the view state of other present users

- Known issue:
    - There is a conflict between double-clicking a contact to open their profile and single-clicking to jump to their view state. I didn't fix this yet because I think the whole interaction is wrong and instead there should be an overlay or context menu to pick between these two options

### CODE REVIEW

File: src/components/Url.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
+import {
```
Comment (geoffreylitt) [inline src/components/Url.ts]: Maybe move this file out of `/components` since it doesn't contain a React component?

Comment (geoffreylitt) [reply src/components/Url.ts]: Actually I'd suggest merging it with `ShareLink.ts` since they seem quite closely related

**Review by geoffreylitt — COMMENTED** [review]
Generally looks great to me, nice work!

One possible issue: in my local testing this branch seems to cause the "white screen of death" sometimes. That may be an unrelated issue though, I'm not sure. Here's a video showing the bug:

https://www.loom.com/share/0f3cc15cfbc1430b94ead39a4b50a9a7

Maybe you could see if you can reproduce that @paulsonnentag ?

Also @pvh do you want to take a look at this PR too

File: src/components/Url.ts (hunk 1 above)
Comment (paulsonnentag) [reply src/components/Url.ts]: Great suggestion, I've combined the two

**Comment by paulsonnentag** [comment]
> One possible issue: in my local testing this branch seems to cause the "white screen of death" sometimes. That may be an unrelated issue though, I'm not sure. Here's a video showing the bug:
> 
> https://www.loom.com/share/0f3cc15cfbc1430b94ead39a4b50a9a7
>

The issue is caused by Arc. If you copy the URL with the CMD+SHIFT+C shortcut the URL is decoded so this:

http://localhost:5173/blutack/?document=web%2Bpushpin%3A%2F%2Ftext%2F830f7ad5-4e08-455e-ad11-3271054678ca

get's turned into this:

http://localhost:5173/blutack/?document=web+pushpin://text/830f7ad5-4e08-455e-ad11-3271054678ca

This is not actually a white screen of death. If you provide no document parameter (or it can't parse the document parameter in this case) it just shows an empty page and you can navigate to a page with the omnibox

**Pull request closed without merging**